File: java/meal-planner-backend/src/main/java/com/meal_planner/meal_planner_backend/repository/MealRepository.java
package com.meal_planner.meal_planner_backend.repository;

import com.meal_planner.meal_planner_backend.model.Day;
import com.meal_planner.meal_planner_backend.model.Meal;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.util.Arrays;

@Repository
public class MealRepository {

    @Autowired
    private Meal meal;

    public Meal getMealPlan() {
        return meal;
    }

    public String getMeal(String day) {
        validateDay(day);
        return meal.getMeal(day);
    }

    public void updateMeal(String day, String mealName) {
        validateDay(day);
        meal.setMeal(day, mealName);
    }

    private void validateDay(String day) {
        boolean isValid = Arrays.stream(Day.values())
                .anyMatch(d -> d.getValue().equals(day));

        if (!isValid) {
            throw new IllegalArgumentException("Invalid day: " + day);
        }
    }
}

File: csharp/MealPlanner.Infrastructure/Repositories/MealRepository.cs
using MealPlanner.Core.Interfaces;
using MealPlanner.Core.Models;
using System.Linq;

namespace MealPlanner.Infrastructure.Repositories;

public class MealRepository : IMealRepository
{
    private readonly Meal _meal = new();

    public Meal GetMealPlan() => _meal;

    public string GetMeal(string day)
    {
        ValidateDay(day);
        return _meal.GetMeal(day);
    }

    public void UpdateMeal(string day, string mealName)
    {
        ValidateDay(day);
        _meal.SetMeal(day, mealName);
    }

    private static void ValidateDay(string day)
    {
        var isValid = Enum.GetValues<Day>().Any(d => d.ToString().ToLower() == day);

        if (!isValid) throw new ArgumentException("Invalid day", nameof(day));
    }
}